### DISCUSSION TITLE
Bug in GeometryValidation for gdf with None area_of_use

### DISCUSSION THREAD

**Discussion by joshuacortez** [thinkingmachines/geowrangler#139] [Usability Testing Q&A] [answered]
Hello,

I ran into an error after running this on a geodataframe of coastal areas [[GDrive Link to gpkg file](https://drive.google.com/file/d/14_dwc9O4DV5ysJmEVjb5pXgSYnDGbWvn/view?usp=sharing)]

```
from geowrangler.validation import GeometryValidation
GeometryValidation(gdf).validate_all()
```

It returned this error 
```
 174 @patch
    175 def check(
    176     self: CrsBoundsValidator,
    177     geometry: BaseGeometry,  # Geometry to validate
    178     gdf: gpd.GeoDataFrame,  # GeoDataframe to check
    179 ) -> bool:
    180     """Checks if polygon is within bounds of crs"""
--> 181     xmin, ymin, xmax, ymax = gdf.crs.area_of_use.bounds
    182     b_xmin, b_ymin, b_xmax, b_ymax = geometry.bounds
    183     return (
    184         (b_xmin >= xmin) and (b_ymin >= ymin) and (b_xmax <= xmax) and (b_ymax <= ymax)
    185     )

AttributeError: 'NoneType' object has no attribute 'bounds'
```

It turns out my gdf has an `area_of_use` that's `None`. This is probably because the geometries were just programmatically generated from shapely objects. Would it be possible for `GeometryValidation` to support geodataframes without a defined `area_of_use`?

Turns out it's just an issue with the `crs_bounds` validator. It worked when I used just the following validators `("null", "self_intersecting", "orientation", "area")`

**Comment by butchtm** [comment]
@tm-kah-alforja @jtmiclat I  suggest tagging this as a bug  so we can improve the validation error message.

**Reply by ghost** [reply]
created issue #139 for this

**Comment by jtmiclat** [comment]
This seems to be a bug with the gpkg. It doesnt have a proper crs set.
```
In [2]: gdf = gpd.read_file("/vagrant/coastal_areas_IDN.gpkg")

In [3]: gdf.crs
Out[3]: 
<Geographic 2D CRS: GEOGCS["Undefined geographic SRS",DATUM["unknown", ...>
Name: Undefined geographic SRS
Axis Info [ellipsoidal]:
- lat[north]: Latitude (degree)
- lon[east]: Longitude (degree)
Area of Use:
- undefined
Datum: unknown
- Ellipsoid: unknown
- Prime Meridian: Greenwich
```

**Comment by jtmiclat** [comment, marked as answer]
Fixed in #142 when crs or area_of_use is not defined, assume all crs bounds is a failed check.
